### PR TITLE
Update find command error messages and tests

### PR DESCRIPTION
**Pull request by jennibearduit**
- Fixed minor error in `MESSAGE_CONSTRAINTS` of `phone` and updated `PHONE_NUMBER` and `NEXT_OF_KIN_PHONE` in UG
- Created a new field called `MESSAGE_CONSTRAINTS_KEYWORD` for fields used in `FindCommand`
- Updated `FindCommandParserTest` to test invalid `Phone` and `CaseNumber` keywords.

### CODE REVIEW

File: src/main/java/seedu/track2gather/model/person/attributes/CaseNumber.java [hunk 1]
```diff
@@ -8,7 +8,9 @@
  */
 public class CaseNumber extends Attribute<String> implements Comparable<CaseNumber> {
     public static final String MESSAGE_CONSTRAINTS =
-            "Case number should be a positive integer with no leading zeros and at most 6 digits";
+            "Case number should be a positive integer with no leading zeros and contain 1 to 6 digits";
+    public static final String MESSAGE_CONSTRAINTS_KEYWORD =
+            "Case number keyword should be a positive integer with no leading zeros and contain 1 to 6 digits";
```
Comment (lrnshk) [inline src/main/java/seedu/track2gather/model/person/attributes/CaseNumber.java:13]: I think `MESSAGE_CONSTRAINTS` should be "Case number should contain only positive integers with no leading zeros, and must be between 1 to 6 digits long." Same for `MESSAGE_CONSTRAINTS_KEYWORD` except with the word 'keyword' in it.

Also, all your instances of `MESSAGE_CONSTRAINTS` and `MESSAGE_CONSTRAINTS_KEYWORD` are missing full stops.

File: src/main/java/seedu/track2gather/model/person/attributes/Name.java [hunk 2]
```diff
@@ -10,6 +10,8 @@
 
     public static final String MESSAGE_CONSTRAINTS =
             "Names should only contain alphanumeric characters and spaces, and it should not be blank";
+    public static final String MESSAGE_CONSTRAINTS_KEYWORD =
+            "Name keywords should only contain alphanumeric characters and spaces, and it should not be blank";
```
Comment (lrnshk) [inline src/main/java/seedu/track2gather/model/person/attributes/Name.java:14]: Your `MESSAGE_CONSTRAINTS_KEYWORD` uses 'keyword' singular, but here you use the 'keywords' plural. Please pick one and standardise

**Review by lrnshk — CHANGES_REQUESTED** [review]
Minor changes

File: src/main/java/seedu/track2gather/model/person/attributes/Name.java (hunk 2 above)
Comment (jennibearduit) [reply src/main/java/seedu/track2gather/model/person/attributes/Name.java:14]: Thanks for the suggestion! I've made the changes.

File: docs/UserGuide.md [hunk 3]
```diff
@@ -86,15 +86,15 @@ Refer to the [Features](#features) below for details of each command.
 Field | Format
 ------|------------------
 `NAME` | Names should only contain alphanumeric characters and spaces, and should not be blank.
-`PHONE_NUMBER` | Phone numbers should only contain numbers and should be at least 3 digits long.
+`PHONE_NUMBER` | Phone numbers should only contain positive integers and should be at least 3 digits long.
```
Comment (lrnshk) [inline docs/UserGuide.md:89]: Wait, is this supposed to be positive integers? I thought phone numbers can take 0 as well.

Comment (lrnshk) [reply docs/UserGuide.md:89]: Also it does not match NOK phone number.

File: src/main/java/seedu/track2gather/model/person/attributes/Period.java [hunk 4]
```diff
@@ -19,6 +19,8 @@
                     + "and must occur earlier than the end date by at least 1 second.";
     public static final String MESSAGE_CONSTRAINTS_DATE =
             "Date should be in the ISO-8601 format (i.e. yyyy-MM-dd)";
```
Comment (lrnshk) [inline src/main/java/seedu/track2gather/model/person/attributes/Period.java:21]: Just wondering, you use plural for `MESSAGE_CONSTRAINTS` for Names and Case numbers and phone numbers etc:

```
public static final String MESSAGE_CONSTRAINTS =
            "Names should only contain alphanumeric characters and spaces, and it should not be blank";
```

But you use 'date' here.

**Review by lrnshk — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/track2gather/model/person/attributes/Period.java (hunk 4 above)
Comment (jennibearduit) [reply src/main/java/seedu/track2gather/model/person/attributes/Period.java]: Yeah I saw inconsistencies between `MESSAGE_CONSTRAINTS` with UG so I fixed `Name` and `CaseNumber`, but left out `Date`. Thanks for pointing it out!

**Review by lrnshk — APPROVED** [review]
LGTM, I'll make some minor changes on my own

**Pull request merged**